src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/MemberMatch/MemberMatchService.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            var searchMatchOnly = results.Results.Where(x => x.SearchEntryMode == ValueSets.SearchEntryMode.Match).ToList();
+            if (searchMatchOnly.Count > 1)
+            {
+                throw new MemberMatchMatchingException(Core.Resources.MemberMatchMultipleMatchesFound);
+            }
+
+            if (searchMatchOnly.Count == 0)
+            {
+                throw new MemberMatchMatchingException(Core.Resources.MemberMatchNoMatchFound);
+            }
+
+            var match = searchMatchOnly[0];
+            var element = _resourceDeserializer.Deserialize(match.Resource);
+            var foundPatient = element.ToPoco<Patient>();
+            var id = foundPatient.Identifier.Where(x => x.Type.Coding.Exists(x => x.Code == "MB")).FirstOrDefault();
+            if (id == null)
+            {
+                throw new MemberMatchMatchingException(Core.Resources.MemberMatchNoMatchFound);
+            }
+
+            var resultPatient = patient.ToPoco<Patient>();
+            var resultId = new Identifier(id.System, id.Value);
+            resultId.Type = new CodeableConcept("http://terminology.hl7.org/CodeSystem/v2-0203", "UMB", "Member Match");
+            resultPatient.Identifier.Add(resultId);
+            var result = resultPatient.ToResourceElement();
+            return result;
+        }
+
+        private Expression CreateSearchExpression(ResourceElement coverage, ResourceElement patient)
+        {
+            var coverageValues = _searchIndexer.Extract(coverage);
+            var patientValues = _searchIndexer.Extract(patient);
+            var expressions = new List<Expression>();
+            var reverseChainExpressions = new List<Expression>();
+
+            foreach (var patientValue in patientValues)
+            {
+                if (IgnoreInSearch(patientValue))
+                {
+                    continue;
+                }
+
+                var modifier = string.Empty;
+                if (patientValue.SearchParameter.Type == ValueSets.SearchParamType.String)
+                {
+                    modifier = ":exact";
+                }
+
+                expressions.Add(_expressionParser.Parse(new[] { "Patient" }, patientValue.SearchParameter.Code + modifier, patientValue.Value.ToString()));
+            }
+
+            foreach (var coverageValue in coverageValues)
+            {
+                if (IgnoreInSearch(coverageValue))
+                {
+                    continue;
+                }
+
+                var modifier = string.Empty;
+                if (coverageValue.SearchParameter.Type == ValueSets.SearchParamType.String)
+                {
+                    modifier = ":exact";
+                }
+
+                reverseChainExpressions.Add(_expressionParser.Parse(new[] { "Coverage" }, coverageValue.SearchParameter.Code + modifier, coverageValue.Value.ToString()));
+            }
+
+            if (reverseChainExpressions.Count != 0)
```

Comment:
nit: Would it be okay fail the request here if `reverseChainExpressions.Count == 0` without executing the search?

Reply:
I think it's ok to do search based on only coverage or patient search indecies.